lib/sidekiq/web.rb

```diff
@@ -41,226 +45,56 @@ def custom_tabs
       end
       alias_method :tabs, :custom_tabs
 
-      attr_accessor :app_url
-    end
-
-    get "/busy" do
-      erb :busy
-    end
-
-    post "/busy" do
-      if params['identity']
-        p = Sidekiq::Process.new('identity' => params['identity'])
-        p.quiet! if params[:quiet]
-        p.stop! if params[:stop]
-      else
-        processes.each do |pro|
-          pro.quiet! if params[:quiet]
-          pro.stop! if params[:stop]
-        end
+      def locales
+        @locales ||= LOCALES
       end
-      redirect "#{root_path}busy"
-    end
-
-    get "/queues" do
-      @queues = Sidekiq::Queue.all
-      erb :queues
-    end
-
-    get "/queues/:name" do
-      halt 404 unless params[:name]
-      @count = (params[:count] || 25).to_i
-      @name = params[:name]
-      @queue = Sidekiq::Queue.new(@name)
-      (@current_page, @total_size, @messages) = page("queue:#{@name}", params[:page], @count)
-      @messages = @messages.map { |msg| Sidekiq::Job.new(msg, @name) }
-      erb :queue
-    end
-
-    post "/queues/:name" do
-      Sidekiq::Queue.new(params[:name]).clear
-      redirect "#{root_path}queues"
-    end
 
-    post "/queues/:name/delete" do
-      Sidekiq::Job.new(params[:key_val], params[:name]).delete
-      redirect_with_query("#{root_path}queues/#{params[:name]}")
-    end
-
-    get '/morgue' do
-      @count = (params[:count] || 25).to_i
-      (@current_page, @total_size, @dead) = page("dead", params[:page], @count, reverse: true)
-      @dead = @dead.map { |msg, score| Sidekiq::SortedEntry.new(nil, score, msg) }
-      erb :morgue
-    end
-
-    get "/morgue/:key" do
-      halt 404 unless params['key']
-      @dead = Sidekiq::DeadSet.new.fetch(*parse_params(params['key'])).first
-      redirect "#{root_path}morgue" if @dead.nil?
-      erb :dead
-    end
-
-    post '/morgue' do
-      redirect request.path unless params['key']
-
-      params['key'].each do |key|
-        job = Sidekiq::DeadSet.new.fetch(*parse_params(key)).first
-        retry_or_delete_or_kill job, params if job
+      def session_secret=(secret)
+        @secret = secret
       end
-      redirect_with_query("#{root_path}morgue")
-    end
 
-    post "/morgue/all/delete" do
-      Sidekiq::DeadSet.new.clear
-      redirect "#{root_path}morgue"
-    end
-
-    post "/morgue/all/retry" do
-      Sidekiq::DeadSet.new.retry_all
-      redirect "#{root_path}morgue"
-    end
-
-    post "/morgue/:key" do
-      halt 404 unless params['key']
-      job = Sidekiq::DeadSet.new.fetch(*parse_params(params['key'])).first
-      retry_or_delete_or_kill job, params if job
-      redirect_with_query("#{root_path}morgue")
+      attr_accessor :app_url, :session_secret
+      attr_writer :locales
     end
 
+    def initialize
+      secret = Web.session_secret
```

Comment:
This requires the user to set the session_secret manually.  Could we pull it out of the Rails app to make Sidekiq work without any user intervention, e.g.

```ruby

def retrieve_secret
  sec = nil
  sec ||= Web.session_secret
  sec ||= Rails.application.session_secret if defined?(::Rails) # not sure of the syntax in Rails 3.2, 4.x, 5.x...
  sec ||= SecureRandom.hex(64)
  sec
end
```